src/core/mount-setup.c

```diff
@@ -405,7 +406,15 @@ int mount_setup(bool loaded_policy) {
          * systemd. */
         mkdir_label("/run/systemd", 0755);
         mkdir_label("/run/systemd/system", 0755);
-        mkdir_label("/run/systemd/inaccessible", 0000);
+        mkdir_label(SD_INACCESSIBLE_PATH, 0755);
+
+        // Set up inaccessible items
+        mknod(SD_INACCESSIBLE_PATH_REG, S_IFREG | 0000, 0);
+        mkdir_label(SD_INACCESSIBLE_PATH_DIR, 0000);
+        mknod(SD_INACCESSIBLE_PATH_BLK, S_IFBLK | 0000, makedev(0, 0));
+        mknod(SD_INACCESSIBLE_PATH_CHR, S_IFCHR | 0000, makedev(0, 0));
+        mknod(SD_INACCESSIBLE_PATH_FIFO, S_IFIFO | 0000, 0);
```

Comment:
i think it would be nicer to use mkfifo(2) for this, simply because it's not any more complex to do, and the man page says we should.